### PR TITLE
Issue #354: users can query a view with different columns in select() and filter()

### PR DESCRIPTION
**Pull request by himanshukohli09**
…olumns in select() and flter().

### CODE REVIEW

File: connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala [hunk 1]
```diff
@@ -163,6 +165,32 @@ class SparkBigQueryEndToEndReadITSuite extends FunSuite
 
   def testsWithReadInFormat(dataSourceFormat: String, dataFormat: String): Unit = {
 
+    test(("testing view with different columns for select and filter. " +
+      "DataSource %s. Data Format %s").format(dataSourceFormat, dataFormat)) {
+
+      for(i <- 1 to 2){
+        var df = spark.read.format(dataSourceFormat)
```
Comment (davidrabinowitz) [inline connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala:172]: please use `val` and add another value for the cached DF

File: connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala [hunk 2]
```diff
@@ -163,6 +165,32 @@ class SparkBigQueryEndToEndReadITSuite extends FunSuite
 
   def testsWithReadInFormat(dataSourceFormat: String, dataFormat: String): Unit = {
 
+    test(("testing view with different columns for select and filter. " +
+      "DataSource %s. Data Format %s").format(dataSourceFormat, dataFormat)) {
+
+      for(i <- 1 to 2){
```
Comment (davidrabinowitz) [inline connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala:171]: can you please move the DF creation to a helper method and remove the loop in order to improve readability?

File: connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala [hunk 3]
```diff
@@ -163,6 +165,32 @@ class SparkBigQueryEndToEndReadITSuite extends FunSuite
...
+          .option("viewMaterializationProject", System.getenv("GOOGLE_CLOUD_PROJECT"))
+          .option("viewMaterializationDataset", testDataset)
+          .option("readDataFormat", dataFormat)
+          .load()
+
+        if(i==2){
+          df = df.cache()
+        }
+
+        val result = df
```
Comment (davidrabinowitz) [inline connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala:184]: Please add comment - should this filter run on the BigQuery side or on the Spark side?

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala (hunk 1 above)
Comment (himanshukohli09) [reply connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala]: Done

File: connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala (hunk 2 above)
Comment (himanshukohli09) [reply connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala]: done

File: connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala (hunk 3 above)
Comment (himanshukohli09) [reply connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndReadITSuite.scala]: done

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**